libs/island-ui/core/src/lib/FilterSearch/FilterSearch.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { Component } from 'react'
+import * as styles from './FilterSearch.treat'
+import cn from 'classnames'
+
+import { Box } from '../Box/Box'
+import { Button } from '../Button/Button'
+import { Icon, IconProps } from '../Icon/Icon'
+import { InputSearch } from '../InputSearch/InputSearch'
+import { Text } from '../Text/Text'
+import { theme } from '@island.is/island-ui/theme'
+import { useIsomorphicLayoutEffect, useWindowSize } from 'react-use'
+
+export type IconVariantTypes = 'default' | 'sidebar'
+
+export type InputValues = {
+  placeholder: string
+  isLoading: boolean
+  colored: boolean
+  value?: string
+  onChange?: (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => void
+}
+export type ClearValues = {
+  text: string
+  onClick?: (event: React.MouseEvent<HTMLSpanElement, MouseEvent>) => void
+}
+
+// 'default': The component will be styled with default styles and all styles
+//            provided in ClassName will be added also.
+// 'noStyles: The component will not use any styles, only styles provided in
+//            ClassName will be used.
+
+export type FilterSearchType = 'default' | 'noStyles'
+
+export interface FilterSearchProps {
+  labelMobileButton: string
+  LabelMobileCloseButton: string
+  LabelMobileResultButton: string
```

Comment:
`LabelMobileCloseButton` and `LabelMobileResultButton` should start with `l` instead of `L`